django/middleware/http.py

```diff
@@ -19,6 +19,9 @@ def process_response(self, request, response):
         if not response.streaming and not response.has_header('Content-Length'):
             response['Content-Length'] = str(len(response.content))
 
+        if request.method not in ('GET', 'HEAD'):
+            return response
+
```

Comment:
Tickets to remove both headers have been accepted, so it shouldn't matter in the end, but for this PR I think it's better to add the header as we had been doing previously, since that aspect of the middleware is completely unrelated to the conditional request processing.